### PR TITLE
Extend forwarded for

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: http-server/src/main/java/com/proofpoint/http/server/CidrSet.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CidrSet
```
Comment (gwittel) [inline http-server/src/main/java/com/proofpoint/http/server/CidrSet.java]: Maybe Cidr4Set to match the v4 content?

Comment (johngmyers) [reply http-server/src/main/java/com/proofpoint/http/server/CidrSet.java]: The interface should be capable of handling IPv6 as well. Notice that `containsAddress` takes an `InetAddress`. It's just that the implementation doesn't currently support IPv6.

File: CHANGES [hunk 2]
```diff
@@ -1,5 +1,9 @@
 Platform 1.74
 
+* We added the "http-server.internal-networks" configuration property.
+  This takes a comma-separated list of IPv4 CIDR blocks which are additionally
+  trusted when processing the X-Forwarded-For: header on incoming requests.
```
Comment (gwittel) [inline CHANGES]: Maybe clarify that this is in addition to the standard private networks and not overriding?

File: http-server/src/main/java/com/proofpoint/http/server/CidrSet.java (hunk 1 above)
Comment (gwittel) [reply http-server/src/main/java/com/proofpoint/http/server/CidrSet.java]: Yeah I saw that.  I'm fine just mentioning the caveat and sorting out v6 handling later.

File: CHANGES (hunk 2 above)
Comment (johngmyers) [reply CHANGES]: It does say "additionally". How would you word this?

Comment (gwittel) [reply CHANGES]: Maybe:  "are trusted in addition to private IP address ranges when ..."

**Comment by johngmyers** [comment]
fixup commits pushed

**Comment by johngmyers** [comment]
Squashed fixup commits and merged

**Pull request closed without merging**